src/operations/generate/remoteGitProjectPersister.ts

```diff
@@ -0,0 +1,58 @@
+import { ActionResult } from "../../action/ActionResult";
+import { logger } from "../../internal/util/logger";
+import { GitCommandGitProject } from "../../project/git/GitCommandGitProject";
+import { GitProject } from "../../project/git/GitProject";
+import { Project } from "../../project/Project";
+import { doWithRetry, RetryOptions } from "../../util/retry";
+import { GitHubRepoRef, isGitHubRepoRef } from "../common/GitHubRepoRef";
+import { ProjectOperationCredentials } from "../common/ProjectOperationCredentials";
+import { isRemoteRepoRef, RepoId } from "../common/RepoId";
+import { ProjectPersister } from "./generatorUtils";
+
+/**
+ * Persist project to GitHub or another remote, returning remote details. Use retry.
+ * @param {Project} p project to persist
+ * @param {ProjectOperationCredentials} creds
+ * @param targetId id of target repo to create
+ * @return {Promise<ActionResult<GitProject>>}
+ */
+export const RemoteGitProjectPersister: ProjectPersister<GitProject> =
+    (p: Project,
+     creds: ProjectOperationCredentials,
+     targetId: RepoId) => {
+        // Default to github.com if we don't have more information
+        const gid = isRemoteRepoRef(targetId) ? targetId : new GitHubRepoRef(targetId.owner, targetId.repo);
+        const gp: GitProject =
+            GitCommandGitProject.fromProject(p, creds);
+        return gp.init()
+            .then(() => {
+                return isGitHubRepoRef(gid) ? gp.configureFromRemote() : {};
```

Comment:
why only configure GitHub ones?

Reply:
Not sure what this means for BitBucket yet